Source/CesiumRuntime/Private/CesiumGltfComponent.cpp

```diff
@@ -2373,12 +2373,12 @@ static void fillTriangles(
 
   triangles.Reserve(triangleCount);
 
-  for (TIndex i = 0; i < static_cast<TIndex>(triangleCount); ++i) {
-    TIndex index0 = 3 * i;
-    triangles.Add(Chaos::TVector<TIndex, 3>(
-        static_cast<TIndex>(indices[index0 + 1]),
-        static_cast<TIndex>(indices[index0]),
-        static_cast<TIndex>(indices[index0 + 2])));
+  for (int32 i = 0; i < triangleCount; ++i) {
+    const int32 index0 = 3 * i;
```

Comment:
This is the key line that fixes the bug. The rest I'm pretty sure is unnecessary (though it won't do any harm). Even if `i` is small enough to fit into a uint16, there is no guarantee that `3 * i` still will be. And when it's not, it will overflow and access the wrong source indices.